docs/source/developers/cpp/building.rst

```diff
@@ -121,6 +121,99 @@ On MSYS2:
 Building
 ========
 
+All the instructions below assume that you have cloned the Arrow git
+repository and navigated to the ``cpp`` subdirectory:
+
+.. code-block:: shell
+
+   $ git clone https://github.com/apache/arrow.git
+   $ cd arrow/cpp
+
+CMake presets
+-------------
+
+Using CMake version 3.21.0 or higher, some presets for various build
+configurations are provided.  You can get a list of the available presets
+using ``cmake --list-presets``:
+
+.. code-block:: shell
+
+   $ cmake --list-presets
+   Available configure presets:
+
+     "ninja-debug-minimal"     - Debug build without anything enabled
+     "ninja-debug-basic"       - Debug build with tests and reduced dependencies
+     "ninja-debug"             - Debug build with tests and more optional components
+      [ etc. ]
+
+You can inspect the specific options enabled by a given preset using
+``cmake -N --preset <preset name>``:
+
+.. code-block:: shell
+
+   $ cmake --preset -N ninja-debug-minimal
+   Preset CMake variables:
+
+     ARROW_BUILD_INTEGRATION="OFF"
+     ARROW_BUILD_STATIC="OFF"
+     ARROW_BUILD_TESTS="OFF"
+     ARROW_EXTRA_ERROR_CONTEXT="ON"
+     ARROW_WITH_RE2="OFF"
+     ARROW_WITH_UTF8PROC="OFF"
+     CMAKE_BUILD_TYPE="Debug"
+
+You can also create a build from a given preset:
+
+.. code-block:: shell
+
+   $ mkdir build
+   $ cd build
+   $ cmake .. --preset ninja-debug-minimal
```

Comment:
Is there a way you can combine a preset with some additional manual config options? (eg in practice you will most of the time also have to specify for example DCMAKE_INSTALL_PREFIX, or for example specifying the dependency source?)

Reply:
One can still do that with presets i.e `cmake .. --preset ninja-debug-minimal -DCMAKE_BUILD_TYPE:STRING=Release` will override build type. Command line options take precendence